RecoJets/JetProducers/python/PileupJetIDParams_cfi.py

```diff
@@ -1,6 +1,212 @@
+                    "pull"      ,
+                    "jetR"      ,
+                    "jetRchg"   ,
+                )
+                ),
+            cms.PSet(
+                jEtaMin = cms.double(2.5),
+                jEtaMax = cms.double(2.75),
+                tmvaWeights   = cms.FileInPath("RecoJets/JetProducers/data/pileupJetId_102X_Eta2p5To2p75_chs_BDT.weights.xml.gz"),
+                tmvaVariables = cms.vstring(
+                    "nvtx"      ,
+                    "dR2Mean"   ,
+                    "nParticles",
+                    "nCharged"  ,
+                    "majW"      ,
+                    "minW"      ,
+                    "frac01"    ,
+                    "frac02"    ,
+                    "frac03"    ,
+                    "frac04"    ,
+                    "ptD"       ,
+                    "beta"      ,
+                    "pull"      ,
+                    "jetR"      ,
+                    "jetRchg"   ,
+                )
+                ),
+            cms.PSet(
+                jEtaMin = cms.double(2.75),
+                jEtaMax = cms.double(3.0),
+                tmvaWeights   = cms.FileInPath("RecoJets/JetProducers/data/pileupJetId_102X_Eta2p75To3p0_chs_BDT.weights.xml.gz"),
+                tmvaVariables = cms.vstring(
+                    "nvtx"      ,
+                    "dR2Mean"   ,
+                    "nParticles",
+                    "nCharged"  ,
+                    "majW"      ,
+                    "minW"      ,
+                    "frac01"    ,
+                    "frac02"    ,
+                    "frac03"    ,
+                    "frac04"    ,
+                    "ptD"       ,
+                    "beta"      ,
+                    "pull"      ,
+                    "jetR"      ,
+                    "jetRchg"   ,
+                )
+                ),
+            cms.PSet(
+                jEtaMin = cms.double(3.),
+                jEtaMax = cms.double(5.),
+                tmvaWeights   = cms.FileInPath("RecoJets/JetProducers/data/pileupJetId_102X_Eta3p0To5p0_chs_BDT.weights.xml.gz"),
+                tmvaVariables = cms.vstring(
+                    "nvtx"      ,
+                    "dR2Mean"   ,
+                    "nParticles",
+                    "majW"      ,
+                    "minW"      ,
+                    "frac01"    ,
+                    "frac02"    ,
+                    "frac03"    ,
+                    "frac04"    ,
+                    "ptD"       ,
+                    "pull"      ,
+                    "jetR"      ,
+                )
+                ),
+        ),
+        tmvaSpectators = cms.vstring(
+            "jetPt" ,
+            "jetEta",
+        ),
+        JetIdParams = full_102x_chs_wp,
+        label = cms.string("full")
+)
+
+###################################################################################################################
+full_94x_chs = cms.PSet(
+        impactParTkThreshold = cms.double(1.),
```

Comment:
is anything changing here other than the training file names and the working point PSet?
If not, a more compact and perhaps more maintainable way would be to use

```python
full_94x_chs = full_102x_chs.clone(JetIdParams = full_94x_chs_wp)
full_94x_chs.trainings[0].tmvaWeights = "RecoJets/JetProducers/data/pileupJetId_94X_Eta0p0To2p5_chs_BDT.weights.xml.gz"
...
```

Having the older IDs listed first may also be more maintainable for cases of future additions.

Reply:
same as previous comment. We are currently using the same values for WP in these years, but they will potentially change in the near future. What is your suggestion?

Reply:
```full_94x_chs = full_102x_chs.clone(JetIdParams = full_94x_chs_wp)``` is applying different working points for 94X vs 102X. The suggestion in the earlier comment simply allows not to repeat what's known to be the same.